### DISCUSSION TITLE
ModuleNotFoundError despite corresponding path being in --paths arg?

### DISCUSSION THREAD

**Discussion by aabirbasu** [pyinstaller/pyinstaller#7201] [Help]
So I want to import a file that's at a location from where I'm already importing a bunch of files, meaning its path is already in the --paths arg and is working all good. The import statement for the new file at this location is throwing a ModuleNotFoundError and I'm running out of ideas as to why. Thanks!

**Comment by bwoodsend** [comment]
What kind of module is it? A top level module or a submodule? Is it something you wrote or you installed with pip?

Generally speaking `--paths` is a false friend. If you ever find yourself thinking you need it - you're doing it wrong.

**Reply by aabirbasu** [reply]
The module is another python script I modified. `--paths` was already in use to import other scripts in from this location, so I shouldn't have had to worry about the --paths arg in the first place, at least that's how I understood it

**Reply by rokm** [reply]
How are you importing this module? Using regular `import mymodule`, or something else?

**Reply by aabirbasu** [reply]
Good ol' `import mymodule`, yes.

**Reply by rokm** [reply]
If it is a direct import like that (from entry-point script or another module that has been imported in an import chain originating from the entry-point script), then it should be picked up. Such imports ending up missing usually indicate that they were not resolvable during the analysis.

In your `build/<name>` directory, there's a file called Analysis-00.toc, which lists all the relevant files that were discovered during analysis - can you check that it contains other modules from that location (the ones that you say are *not* problematic)? And then check that the offending module is not listed in the file?

In the `build/name` directory, there is also an `xref-<name>.html` file; can you open it in the web browser, and search for the offending module there? Is it marked as `MissingModule`?

For any further diagnosis though, you'll probably need to provide us a basic example with the module/script layout that reproduces the problem.
